packages/cli/src/commands/publish/workflow.ts

```diff
@@ -0,0 +1,68 @@
+import { WorkflowRepository } from '@n8n/db';
+import { Command } from '@n8n/decorators';
+import { Container } from '@n8n/di';
+import { z } from 'zod';
+
+import { BaseCommand } from '../base-command';
+
+const flagsSchema = z.object({
+	id: z.string().describe('The ID of the workflow to publish'),
+	versionId: z
+		.string()
+		.describe('The version ID to publish. If not provided, publishes the current version')
+		.optional(),
+	all: z.boolean().describe('(Deprecated) This flag is no longer supported').optional(),
+});
+
+@Command({
+	name: 'publish:workflow',
+	description:
+		'Publish a specific version of a workflow. If no version is specified, publishes the current version.',
+	examples: ['--id=5 --versionId=abc123', '--id=5'],
+	flagsSchema,
+})
+export class PublishWorkflowCommand extends BaseCommand<z.infer<typeof flagsSchema>> {
+	async run() {
+		const { flags } = this;
+
+		// Educate users who try to use --all flag
+		if (flags.all) {
+			this.logger.error('The --all flag is no longer supported for workflow publishing.');
+			this.logger.error(
+				'Please publish workflows individually using: publish:workflow --id=<workflow-id> [--versionId=<version-id>]',
+			);
+			return;
+		}
+
+		if (!flags.id) {
```

Comment:
This check can be removed. Zod validates if the field `id` is present and if not, execution will not reach this method